### PR TITLE
Metric types updates

### PR DESCRIPTION
**Pull request by bradleyjay**
### What does this PR do?
Update to the Metric Type documentation - reorganized for clarity, tweaked the examples, additional explanation for certain Metric Type behaviors added.

Specifically, we're delineating the difference between what's submitted to Datadog, and what is stored. We aggregate types differently, but only through the Agent and DogstatsD. It's important to clarify this, as many users don't know the difference between aggregation at submission (which affects the data stored) and aggregation at query time (which does not).

### Motivation
This document is referenced quite frequently - there's a little more detail we could provide for customers and the Support team, both of which read this daily.

### Preview link

* https://docs-staging.datadoghq.com/bradleyjay/metricstypes/developers/metrics/types/?tab=count#metric-submission-types

### CODE REVIEW

File: content/en/developers/metrics/metrics_type.md [hunk 1]
```diff
@@ -84,22 +91,25 @@ Discover how to submit count metrics:
 {{% /tab %}}
 {{% tab "RATE" %}}
 
-**The `RATE` metric submission type represents the number of events over a defined time interval (flush interval) that's normalized per-second.** A `RATE` can be used to track the rate of requests hitting your webservers.
+**The `RATE` metric submission type represents the total number of event occurences per second.** A `RATE` is used to track how often something is happening - the frequency of connections made to a database, or the flow of requests made to an endpoint.
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
**The `RATE` metric submission type represents the total number of event occurences per second.** A `RATE` is used to track how often something is happening—like the frequency of connections made to a database, or the flow of requests made to an endpoint.
```

File: content/en/developers/metrics/metrics_type.md [hunk 2]
```diff
@@ -111,15 +121,25 @@ Discover how to submit rate metrics:
 {{% /tab %}}
 {{% tab "GAUGE" %}}
 
-**The `GAUGE` metric submission type represents a value of a particular thing reporting continuously over time.** It is a snapshot of the last recorded value of a particular thing during a defined time interval (flush interval). A `GAUGE` can be used to represent the temperature or memory usage.
+**The `GAUGE` metric submission type represents a snapshot of one time interval.**  This representative snapshot value is the last value submitted to the Agent during the time interval.
+
+`GAUGE` metrics are ideal for taking a measure of something reporting continuously. Suppose a webserver is running the Datadog agent, tracking the request latency in milliseconds. If each request has a latency of 300ms, checking request latency three times in one time interval should not report 900ms - the `GAUGE` would report 300ms.
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
`GAUGE` metrics are ideal for taking a measure of something reporting continuously. Suppose a webserver is running the Datadog Agent, tracking the request latency in milliseconds. If each request has a latency of 300ms, checking request latency three times in one time interval should not report 900ms—instead, the `GAUGE` would report 300ms.
```

File: content/en/developers/metrics/metrics_type.md [hunk 3]
```diff
@@ -131,26 +151,28 @@ Discover how to submit gauge metrics:
 [2]: /developers/metrics/dogstatsd_metrics_submission/#gauge
 [3]: /api/?lang=python#post-timeseries-points
 {{% /tab %}}
+<!--
 {{% tab "HISTOGRAM" %}}
 
-**The `HISTOGRAM` metric submission type allows you to measure the statistical distribution of a set of values**. Datadog's `HISTOGRAM` metric type is an extension of the [StatsD timing metric type][1]: it aggregates (on the Agent-side) the values that are sent during a defined time interval (the default flush interval is 10s) and produces different timeseries representing the different aggregations possible for the set of values. Depending on the aggregation, the metric type stored by Datadog is different.
+**The `HISTOGRAM` metric submission type represents a statistical distribution calculated at the Agent of a set of values in one time interval**. `HISTOGRAM` is stored in-app as 5 different timeseries representing the distribution of the set of values in that time interval. It is an extension of the [StatsD timing metric type][1]. A `HISTOGRAM` can be used to represent response time on a host or total event duration on a host.
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
**The `HISTOGRAM` metric submission type represents a statistical distribution calculated at the Agent of a set of values in one time interval**. `HISTOGRAM` is stored in-app as five different timeseries representing the distribution of the set of values in that time interval. It is an extension of the [StatsD timing metric type][1]. A `HISTOGRAM` can be used to represent response time on a host or total event duration on a host.
```

File: content/en/developers/metrics/metrics_type.md [hunk 4]
```diff
@@ -131,26 +151,28 @@ Discover how to submit gauge metrics:
...
 
-For example: if you send `X` values for a `HISTOGRAM` metric `<METRIC_NAME>` during an Agent flush interval, the following timeseries are produced by the Agent by default:
+For example: if you send `X` values for a `HISTOGRAM` metric `<METRIC_NAME>` during one time interval, the Agent aggregates the set of values to produce the following 5 representative timeseries by default:
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
For example: if you send `X` values for a `HISTOGRAM` metric `<METRIC_NAME>` during one time interval, the Agent aggregates the set of values to produce the following five representative timeseries by default:
```

File: content/en/developers/metrics/metrics_type.md [hunk 5]
```diff
@@ -131,26 +151,28 @@ Discover how to submit gauge metrics:
...
+| `<METRIC_NAME>.95percentile` | Gives you the 95th percentile of those `X` values in the time interval.                                                                                  | GAUGE               |
+| `<METRIC_NAME>.max`          | Gives you the maximum value of those `X` values sent in the time interval.                                                                           | GAUGE               |
 
+For example, suppose a webserver is running the Datadog Agent, and you're tracking the response times received as a `HISTOGRAM` metric. In the one time interval, the webserver receives the set of values [1,1,1,2,2,2,3,3]. 
 
-For instance, say that the `request.response_time.histogram` metric is reported to Datadog through an Agent with the `HISTOGRAM` type for `server:web_1` with the values [1,1,1,2,2,2,3,3] during a flush interval. The following metrics would have then be submitted to Datadog over this flush interval:
+The Agent aggregates the set of values to create the 5 timeseries representing the distribution of the set of values [1,1,1,2,2,2,3,3]. The 5 timeseries are submitted to Datadog and stored as the following in-app metric types:
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
The Agent aggregates the set of values to create the five timeseries representing the distribution of the set of values [1,1,1,2,2,2,3,3]. The five timeseries are submitted to Datadog and stored as the following in-app metric types: 
```

File: content/en/developers/metrics/metrics_type.md [hunk 6]
```diff
@@ -178,45 +200,50 @@ Discover how to submit histogram metrics:
 This feature is in beta. <a href="https://docs.datadoghq.com/help/">Contact Datadog support</a> to enable distribution metrics for your account.
 </div>
 
-**A `DISTRIBUTION` is a metric submission type that allows you to aggregate values sent from multiple hosts during a flush interval to measure statistical distributions across your entire infrastructure.** `DISTRIBUTION` metrics are designed to instrument logical objects, like services, independently from the underlying hosts.
+**The `DISTRIBUTION` metric submission type represents a statistical distribution calculated over your entire infrastructure of a set of values in one time interval.** A `DISTRIBUTION` metric allows you to aggregate values sent from multiple hosts over one time interval.  A `DISTRIBUTION` can be used to represent request latency for your entire infrastructure or customer basket size.
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
**The `DISTRIBUTION` metric submission type represents a statistical distribution calculated over your entire infrastructure of a set of values in one time interval.** A `DISTRIBUTION` metric allows you to aggregate values sent from multiple hosts over one time interval. A `DISTRIBUTION` can be used to represent request latency for your entire infrastructure or customer basket size. 
```

File: content/en/developers/metrics/metrics_type.md [hunk 7]
```diff
@@ -178,45 +200,50 @@ Discover how to submit histogram metrics:
...
 
-Unlike the `HISTOGRAM` metric type, which aggregates on the Agent side during the flush interval, a `DISTRIBUTION` metric sends all the raw data during a flush interval to Datadog, and aggregations occur server-side. Because the underlying data structure represents raw, unaggregated data, distributions provide two major features:
+Unlike the `HISTOGRAM` metric type, which aggregates on the Agent side during one time interval, a `DISTRIBUTION` metric sends the raw data during that time interval to Datadog, and aggregations occur server-side. Because the underlying data structure represents raw, unaggregated data, distributions provide two major features:
 
 * Calculation of percentile aggregations
 * Customization of tagging
 
-For example: if you send `X` values for a `DISTRIBUTION` metric `<METRIC_NAME>` during an interval, the following timeseries are available to query by default:
+For example, if you send `X` values for a `DISTRIBUTION` metric `<METRIC_NAME>` during an interval, the following 5 timeseries are available to query by default:
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
For example, if you send `X` values for a `DISTRIBUTION` metric `<METRIC_NAME>` during an interval, the following five timeseries are available to query by default:
```

File: content/en/developers/metrics/metrics_type.md [hunk 8]
```diff
@@ -178,45 +200,50 @@ Discover how to submit histogram metrics:
...
 
-Additional percentile aggregations (`p50`, `p75`, `p90`, `p95`, `p99`) can be added to distribution metrics. That is, for a distribution metric with added percentile aggregations during a 10 second flush interval, the following aggregations are available: `count`, `sum`, `min`, `max`, `avg`, `p50`, `p75`, `p90`, `p95`, and `p99`.
+Additional percentile aggregations (`p50`, `p75`, `p90`, `p95`, `p99`) can be added to distribution metrics such that the following aggregations are available: `count`, `sum`, `min`, `max`, `avg`, `p50`, `p75`, `p90`, `p95`, and `p99`.
 
-If you were to add percentile aggregations to your distribution metric (as shown in-app [Datadog Distribution Metric page][2]), the following timeseries would be additionally created:
+If you were to add percentile aggregations to your distribution metric (as shown in-app [Datadog Distribution Metric page][2]), the 5 following timeseries would be additionally created:
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
If you were to add percentile aggregations to your distribution metric (as shown in-app [Datadog Distribution Metric page][2]), the five following timeseries would be additionally created:
```

File: content/en/developers/metrics/metrics_type.md [hunk 9]
```diff
@@ -250,24 +277,26 @@ Discover how to submit set metrics:
...
-Datadog accepts metrics submitted from a variety of sources:
+Datadog accepts metrics submitted from three sources:
 
-* [Datadog API][3]
-* [DogStatsD][4]
 * [Agent Check][5]
+* [DogStatsD][4]
+* [Datadog API][3]
+
+These sources submit data as various submission types, which map to the three canonical in-app storage types in Datadog - `COUNT`, `RATE`, and `GAUGE`:
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]:
```suggestion
These sources submit data as various submission types, which map to the three canonical in-app storage types in Datadog—`COUNT`, `RATE`, and `GAUGE`:
```

File: content/en/developers/metrics/metrics_type.md [hunk 10]
```diff
@@ -131,26 +151,28 @@ Discover how to submit gauge metrics:
 [2]: /developers/metrics/dogstatsd_metrics_submission/#gauge
 [3]: /api/?lang=python#post-timeseries-points
 {{% /tab %}}
+<!--
```
Comment (cswatt) [inline content/en/developers/metrics/metrics_type.md]: Is the following section supposed to be commented out?

File: content/en/developers/metrics/types.md [hunk 11]
```diff
@@ -31,47 +33,53 @@ The following metric submission types are accepted:
 * [SET](?tab=set#metric-type-definition)
 * [TIMERS](?tab=timers#metric-type-definition)
 
-Once a metric is submitted, there are 3 metric storage types within the Datadog web application:
+When data is received, the submitted metric type is mapped to one of three in-app **Metric Types**. Datadog stores metric data under a few canonical types:
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
When data is received, the submitted metric type is mapped to one of three in-app **Metric Types**. Datadog stores metric data under the following canonical types:
```

File: content/en/developers/metrics/types.md [hunk 12]
```diff
@@ -31,47 +33,53 @@ The following metric submission types are accepted:
...
 
 * `COUNT`
 * `RATE`
 * `GAUGE`
 
-Refer to the [Metric Type Definition](#metric-type-definition) section below to learn more about the different metrics types available. Also, refer to the [Submission Types and Datadog In-App Types Section](#submission-types-and-datadog-in-app-types) to see how each metric type behaves between its submission and its storage within Datadog.
+A metric's type can be viewed on the details sidepanel on the [Metrics Summary page][2]. The metric type can be modified in this side panel; however, this should be done deliberately with care and only if necessary. Modifying metric type *can* change how functions and modifiers interact with your data, potentially affecting graphs and monitors.
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
A metric's type can be viewed on the details sidepanel on the [Metrics Summary page][2]. The metric type can be modified in this side panel. However, this should be done carefully and only if necessary. Modifying metric type *can* change how functions and modifiers interact with your data, potentially affecting graphs and monitors.
```

File: content/en/developers/metrics/types.md [hunk 13]
```diff
@@ -31,47 +33,53 @@ The following metric submission types are accepted:
...
-## Metric type definition
+Data is submitted to Datadog in three main ways:
 
-To better understand the different metrics types, what they represent, and how to modify them within Datadog, consider the following example:
+* [Agent Check][5]
+* [DogStatsD][4]
+* [Datadog API][3]
 
-You have two web servers: `server:web_1` and `server:web_2`. Both web servers continuously receive:
+Most of the data received by Datadog is submitted via the Agent. For data submitted through an Agent Check or DogstatsD, aggregation occur when multiple points arrive in a short time interval. During that time interval, the Agent combines values that belong in the same time series (i.e. values with identical tags) and sends a single representative value for that interval. This combined value is stored with a single timestamp. The way in which values are combined and aggregated varies, depending on metric type.
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
Most of the data received by Datadog is submitted via the Agent. For data submitted through an Agent Check or DogStatsD, aggregation occurs when multiple points arrive in a short time interval. During that time interval, the Agent combines values that belong in the same time series (i.e. values with identical tags) and sends a single representative value for that interval. This combined value is stored with a single timestamp. The way in which values are combined and aggregated varies, depending on metric type.
```

File: content/en/developers/metrics/types.md [hunk 14]
```diff
@@ -31,47 +33,53 @@ The following metric submission types are accepted:
...
 
-* 10 HTTP requests for the first 30 seconds, then
-* 20 HTTP requests for the next 30 seconds, then
-* 0 HTTP requests for the next 30 seconds.
+Data submitted directly to the Datadog API is not aggregated by Datadog before storage (except in the case of Distributions). The raw values sent to Datadog are stored as-is.
 
-### Metric submission types
+### Metric Submission Types
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
### Metric submission types
```

we sentence-case for h3

File: content/en/developers/metrics/types.md [hunk 15]
```diff
@@ -31,47 +33,53 @@ The following metric submission types are accepted:
...
+This table discusses each Metric Submission Type (through Agent or DogStatsD) and how it is aggregated before being mapped into in-app Metric Type. **Note**: no aggregation occurs for API metric submissions.
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
This table discusses each metric submission type (through Agent or DogStatsD) and how it is aggregated before being mapped to an in-app metric type. **Note**: no aggregation occurs for API metric submissions.
```

File: content/en/developers/metrics/types.md [hunk 16]
```diff
@@ -19,9 +19,11 @@ further_reading:
   text: "Official and Community created API and DogStatsD client libraries"
 ---
 
-A metric's type affects how the given metric is aggregated in queries and values displayed in graph visualizations within Datadog. Different metric types can have different default time aggregation functions and different [Metric Type modifiers][1]. A metric's type is displayed on the details sidepanel for the given metric on the [Metrics Summary page][2]. Note: Changing the metric type in this details sidepanel can potentially change metric behavior in all existing visualizations and monitors; potentially rendering historical data as nonsensical.
+Each metric submitted to Datadog has a “Metric Type.” Metric Type affects how the metric is stored, queried, and graphed, as well as how the metric interacts with additional [Metric Type modifiers][1] and functions.
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
Each metric submitted to Datadog has a **metric type**. Metric type affects how the metric is stored, queried, and graphed, as well as how the metric interacts with additional [metric type modifiers][1] and functions.
```

File: content/en/developers/metrics/types.md [hunk 17]
```diff
@@ -19,9 +19,11 @@ further_reading:
...
 
-The following metric submission types are accepted:
+## Metric Types: Storage vs. Submission
+
+Datadog accepts a wide range of **Metric Submission Types**:
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
Datadog accepts a wide range of **metric submission types**:
```

File: content/en/developers/metrics/types.md [hunk 18]
```diff
@@ -85,22 +93,25 @@ Discover how to submit count metrics:
...
-**Note**: A `RATE` is different from the `COUNT` metric submission type, which represents the number of events in the flush interval.
+For example, suppose a webserver is running the Datadog Agent, and you're tracking the number of HTTP requests received as a `RATE` metric. This webserver receives:
 
-For instance, say that the `number.of.requests.rate` metric is reported every 30 seconds to Datadog with the `RATE` type in `server:web_1`.
+* `30` requests in the first 10 seconds
+* `70` requests in the second interval of 10 seconds
+* `50` requests in the third interval of 10 seconds
 
-Each value/data point represents the "rate" of requests. The metric then reports the following values:
+The Agent sums up the requests received in a given interval, then divides by the length of that interval. The Agent then reports the following normalized per-second values:
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
The Agent sums the requests received in a given interval, then divides by the length of that interval. The Agent then reports the following normalized per-second values:
```

File: content/en/developers/metrics/types.md [hunk 19]
```diff
@@ -112,15 +123,25 @@ Discover how to submit rate metrics:
 {{% /tab %}}
 {{% tab "GAUGE" %}}
 
-**The `GAUGE` metric submission type represents a value of a particular thing reporting continuously over time.** It is a snapshot of the last recorded value of a particular thing during a defined time interval (flush interval). A `GAUGE` can be used to represent the temperature or memory usage.
+**The `GAUGE` metric submission type represents a snapshot of one time interval.**  This representative snapshot value is the last value submitted to the Agent during the time interval.
+
+`GAUGE` metrics are ideal for taking a measure of something reporting continuously. Suppose a webserver is running the Datadog Agent, tracking the request latency in milliseconds. If each request has a latency of 300ms, checking request latency three times in one time interval should not report 900ms—instead, the `GAUGE` would report 300ms.
```
Comment (cswatt) [inline content/en/developers/metrics/types.md]:
```suggestion
`GAUGE` metrics are ideal for taking a measure of something reporting continuously. Suppose a webserver is running the Datadog Agent, tracking the request latency in milliseconds. If each request has a latency of 300ms, checking request latency three times in one time interval should not report 900ms—instead, the `GAUGE` reports 300ms.
```

**Comment by l0k0ms** [comment]
Closing this PR and in favor of https://github.com/DataDog/documentation/pull/6302 that fixes all conflicts + finish the re-factor of Histogram/distribution metrics.

**Pull request closed without merging**